datafusion/common/src/config.rs

```diff
@@ -184,6 +184,10 @@ config_namespace! {
         /// Default value for `format.has_header` for `CREATE EXTERNAL TABLE`
         /// if not specified explicitly in the statement.
         pub has_header: bool, default = false
+
+        /// Default value for `format.newlines_in_values` for `CREATE EXTERNAL TABLE`
+        // if not specified explicitly in the statement.
```

Comment:
We need more descriptive comment help for users without context to understand this config option, which should include 1. It's CSV specific 2. Can be overridden by the same config field in `CsvOptions` 3. Its behavior as stated in `CsvOptions`'s comment.

And update in https://github.com/apache/datafusion/blob/main/docs/source/user-guide/configs.md

Reply:
I've expanded the documentation. I forgot about the user guide so I'll do that now as well.